clippy_lints/src/methods/option_take_on_temporary.rs

```diff
@@ -1,27 +1,29 @@
-use clippy_utils::diagnostics::span_lint;
+use clippy_utils::diagnostics::span_lint_and_sugg;
+use clippy_utils::source::snippet_with_applicability;
 use clippy_utils::ty::is_type_diagnostic_item;
+use rustc_errors::Applicability;
 use rustc_hir::Expr;
 use rustc_lint::LateContext;
 use rustc_span::sym;
 
 use super::OPTION_TAKE_ON_TEMPORARY;
 
-pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>, _recv: &'tcx Expr<'_>) {
+pub(super) fn check<'tcx>(cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>, recv: &'tcx Expr<'_>) {
     // Checks if expression type is equal to sym::Option and if the expr is not a syntactic place
     if is_expr_option(cx, expr) & !expr.is_syntactic_place_expr() {
-        span_lint(cx, OPTION_TAKE_ON_TEMPORARY, expr.span, "Format test");
+        span_lint_and_sugg(
+            cx,
+            OPTION_TAKE_ON_TEMPORARY,
+            expr.span,
+            "Called `Option::take()` on a temporary value",
+            "try",
+            format!(
+                "{}",
+                snippet_with_applicability(cx, recv.span, "..", &mut Applicability::MachineApplicable)
```

Comment:
The idea of this function is that you can have a `let mut app = Applicability::MachineApplicable;` that you reference here (`&mut app`) and then use in the `span_lint_and_sugg` call.